### PR TITLE
Utility

### PR DESCRIPTION
**Pull request by daemitus**
This is all miscellaneous trash and bug fixes. Each commit is rather self-contained, it may be easier to view each one individually. 
I need the utility stuff in to facilitate further PRs.

The changes to make note of: 
 - Change the Assembly.Location/CodeBase hooks due to dependencies of plugins not finding themselves. (reloaded)
 - Remove AssemblyLocation from PluginInterface ctor. The above hooks work, its not needed and never made it to prod.
 - Utility class and namespace, all the random extensions got moved here.
 - Handle open handle error when first creating the log rotation file. We discussed this earlier today.
 - GetIcon should be uint, add more overloads for HQ icons. Add a cast, same as before.
 - Removed some methods that were marked as obsolete and already had replacements.

### CODE REVIEW

File: Dalamud/EntryPoint.cs [hunk 1]
```diff
@@ -142,7 +142,7 @@ private static void CullLogFile(string logPath, string? oldPath, int cullingFile
                     var oldFile = new FileInfo(oldPath);
 
                     if (!oldFile.Exists)
-                        oldFile.Create();
+                        oldFile.Create().Close();
```
Comment (daemitus) [inline Dalamud/EntryPoint.cs:145]: Here's your bugfix

File: Dalamud/Game/Internal/Resource/ResourceManager.cs [hunk 2]
```diff
@@ -1,159 +0,0 @@
-using System;
```
Comment (daemitus) [inline Dalamud/Game/Internal/Resource/ResourceManager.cs:1]: You said to delete this

File: Dalamud/Game/Internal/Resource/ResourceManagerAddressResolver.cs [hunk 3]
```diff
@@ -1,28 +0,0 @@
-using System;
```
Comment (daemitus) [inline Dalamud/Game/Internal/Resource/ResourceManagerAddressResolver.cs:1]: and this

File: Dalamud/Logging/PluginLog.cs [hunk 4]
```diff
@@ -8,6 +8,131 @@ namespace Dalamud.Logging
     /// </summary>
     public static class PluginLog
     {
+        #region "Log" prefixed Serilog style methods
```
Comment (daemitus) [inline Dalamud/Logging/PluginLog.cs:11]: When recompiling plugins after removing this earlier, these were used quite a lot. So back in they go.

File: Dalamud/Game/Position3.cs [hunk 5]
```diff
@@ -23,6 +23,19 @@ public struct Position3
         /// </summary>
         public float Y;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Position3"/> struct.
+        /// </summary>
+        /// <param name="x">The X position.</param>
+        /// <param name="z">The Z position.</param>
+        /// <param name="y">The Y position.</param>
+        public Position3(float x, float z, float y)
```
Comment (goaaats) [inline Dalamud/Game/Position3.cs:32]: Marked for deletion

**Review by goaaats — APPROVED** [review]

**Pull request merged**